ground/src/main/java/com/google/android/ground/ui/surveyselector/SurveySelectorViewModel.kt

```diff
@@ -19,51 +19,60 @@ import androidx.lifecycle.LiveData
 import androidx.lifecycle.LiveDataReactiveStreams
 import com.google.android.ground.model.Survey
 import com.google.android.ground.repository.SurveyRepository
-import com.google.android.ground.rx.Loadable
 import com.google.android.ground.system.auth.AuthenticationManager
 import com.google.android.ground.ui.common.AbstractViewModel
 import io.reactivex.Single
 import javax.inject.Inject
-import timber.log.Timber
 
 /** Represents view state and behaviors of the survey selector dialog. */
 class SurveySelectorViewModel
 @Inject
 internal constructor(
   private val surveyRepository: SurveyRepository,
-  authManager: AuthenticationManager
+  private val authManager: AuthenticationManager
 ) : AbstractViewModel() {
-  val surveySummaries: LiveData<Loadable<List<Survey>>?>
-  val offlineSurveys: Single<List<Survey>>
-    get() = surveyRepository.offlineSurveys
 
-  /**
-   * Triggers the specified survey to be loaded and activated.
-   *
-   * @param idx the index in the survey summary list.
-   */
-  fun activateSurvey(idx: Int) {
-    val surveys: List<Survey> = surveySummaries.value?.value()?.orElse(listOf()) ?: listOf()
-    if (surveys.isEmpty()) {
-      Timber.e("Can't activate survey before list is loaded")
-      return
-    }
-    if (idx >= surveys.size) {
-      Timber.e("Can't activate survey at index $idx, only ${surveys.size} surveys in list")
-      return
-    }
-    val (id) = surveys[idx]
-    surveyRepository.activateSurvey(id)
-  }
+  private var attemptingToActivate: String = ""
 
-  fun activateOfflineSurvey(surveyId: String?) {
-    surveyRepository.activateSurvey(surveyId!!)
-  }
+  val surveyActivated: LiveData<Boolean>
+  val surveySummaries: LiveData<List<SurveyItem>>
 
   init {
     surveySummaries =
       LiveDataReactiveStreams.fromPublisher(
-        surveyRepository.getSurveySummaries(authManager.currentUser)
+        offlineSurveys
+          .flatMap { offlineSurveys: List<Survey> ->
+            allSurveys.map { allSurveys: List<Survey> ->
+              allSurveys.map { createSurveyItem(it, offlineSurveys) }
+            }
+          }
+          .toFlowable()
       )
+    surveyActivated =
+      LiveDataReactiveStreams.fromPublisher(
```

Comment:
Can we simplify how we model the survey activation flow? At first glance it seems like we shouldn't be modeling events like `attemptingToActivate` and `surveyActivated` as UI states. Maybe coroutines can help here?

Reply:
Thanks for flagging this. Will try to come up with a more elegant way of doing it.

Reply:
There seems to be some problem while using nested corountine. I converted the Rx code to synchronous by wrapping it inside the following code block. But, the code never reaches the next line. 

```
externalScope.launch {
      withContext(ioDispatcher) {
...
   }
}
```

Reply:
I refactored some code to prevent nesting of corountines and it did work.

Reply:
An alternate solution could be to just set the active survey id and then switch to home screen fragment. Then we won't have to implement a progress dialog for Survey selector fragment either. Thoughts? @gino-m

Reply:
Reverted changes to SurveyRepository